packages/labs/context/src/test/late-provider_test.ts

```diff
@@ -300,4 +301,90 @@ suiteSkipIE('late context provider', () => {
+    const context = createContext<string>(Symbol());
+    @customElement('context-consumer-4')
+    class ContextConsumer4Element extends LitElement {
+      consume = new ContextConsumer(this, {
+        context,
+        subscribe: true,
+        callback: (value) => {
+          this.value = value;
+          this.callCount++;
+        },
+      });
+
+      value = 'consumer initializer';
+
+      callCount = 0;
+    }
+    @customElement('context-provider-grandparent')
+    class ContextProviderGrandparentElement extends LitElement {
+      provide = new ContextProvider(this, {
+        context,
+        initialValue: 'grandparent initial value',
+      });
+    }
+
+    container.innerHTML = `
+      <context-provider-grandparent>
+        <context-consumer-4></context-consumer-4>
+        <late-context-provider-4>
+          <context-consumer-4></context-consumer-4>
+        </late-context-provider-4>
+      </context-provider-grandparent>
+    `;
+    const directChildConsumer = container.querySelector(
+      'context-provider-grandparent > context-consumer-4'
+    ) as ContextConsumer4Element;
+    const indirectChildConsumer = container.querySelector(
+      'late-context-provider-4 > context-consumer-4'
+    ) as ContextConsumer4Element;
+    const grandparentProvider = container.querySelector(
+      'context-provider-grandparent'
+    ) as ContextProviderGrandparentElement;
+
+    await directChildConsumer.updateComplete;
+    assert.equal(directChildConsumer.value, 'grandparent initial value');
+    assert.equal(directChildConsumer.callCount, 1);
+    assert.equal(indirectChildConsumer.value, 'grandparent initial value');
+    assert.equal(indirectChildConsumer.callCount, 1);
+    grandparentProvider.provide.setValue('grandparent updated');
+    await directChildConsumer.updateComplete;
+    assert.equal(directChildConsumer.value, 'grandparent updated');
+    assert.equal(directChildConsumer.callCount, 2);
+    assert.equal(indirectChildConsumer.value, 'grandparent updated');
+    assert.equal(indirectChildConsumer.callCount, 2);
+
+    @customElement('late-context-provider-4')
+    class LateContextProvider4Element extends LitElement {
+      provide = new ContextProvider(this, {
+        context,
+        initialValue: 'late provider initial value',
+      });
+    }
+
+    // The indirect child now gets the late provider's initial value
+    await directChildConsumer.updateComplete;
+    assert.equal(directChildConsumer.value, 'grandparent updated');
+    assert.equal(indirectChildConsumer.value, 'late provider initial value');
+
+    // Updating the middle provider updates its childd, but not its sibling,
+    // the direct child.
+    const middleProvider = container.querySelector(
+      'late-context-provider-4'
+    ) as LateContextProvider4Element;
+    middleProvider.provide.setValue('late provider updated');
+    await directChildConsumer.updateComplete;
+    // bad!
```

Comment:
bad?

Reply:
Sorry that was kept in from earlier assertions, this is correct behavior. directChildConsumer is directly under grandparent, so it still see grandparent updated, but the indirect consumer sees values from the late provider